### DISCUSSION TITLE
prop 'element' required for Combobox in unit tests after upgrading @twilio-paste/core

### DISCUSSION THREAD

**Discussion by mirjammalksoo** [twilio-labs/paste#1819] [Q&A] [answered]
Hey team!

We are in the process of upgrading our `@twilio-paste/core` version to the latest one (from 9.2.1 to 10.10.0).

After upgrading and running our unit tests we now get the following error in every component test that includes Combobox:

```
Warning: Failed prop type: The prop `element` is marked as required in `InputBox`, but its value is `undefined`.
        in InputBox (created by Combobox)
        in Combobox
        in Unknown

      43 |     const [message] = args;
      44 |     if (message.includes('Failed prop type:')) {
    > 45 |         throw new FailedPropTypeError(message);
         |               ^
      46 |     } else if (!shouldShowMessage(message)) {
      47 |         error(...args);
      48 |     }
```

Could it be that the default `element` prop value isn't currently set for Combobox https://github.com/twilio-labs/paste/commit/2eb6dd74eb1cdc06079e64a711e71b93d4f9c823 (like it is for TextArea https://github.com/twilio-labs/paste/commit/0347aef6e8100f9dbc2bcd8830dd5c81b4de5a03)? 

Or has the element prop now become required for Combobox? I currently couldn't find anything about this change from Paste docs. 

Thank you in advance for your help! :)

**Comment by SiTaggart** [comment]
First question I have is, is core and icons the only Paste dependencies you have? If you have others listed they might be pinning combo box to an older version.

The change to combo box was released in 10.10 of core https://github.com/twilio-labs/paste/releases/tag/%40twilio-paste%2Fcore%4010.10.0

You can run `yarn why @twilio-paste/combobox` to find out the resolved version. (Not sure what the npm version is)

I will note that it's weird that Combobox was versioned as a new major. It should have been a patch.

**Reply by mirjammalksoo** [reply]
Yes core and icons are the only ones, icons versions is 5.3.2.

Looking at the yarn.lock, these are the combobox versions included in the file and in core dependencies:
```
 "@twilio-paste/combobox" "^8.0.0"
 "@twilio-paste/combobox-primitive" "^0.3.0"
```

Running `yarn why @twilio-paste/combobox` gives me the same version - `=> Found "@twilio-paste/combobox@8.0.0"`. 

I might be completely wrong but had a hunch that the issue might be that while other components are extending `Pick<BoxProps, 'element'> `https://github.com/twilio-labs/paste/commit/0347aef6e8100f9dbc2bcd8830dd5c81b4de5a03#diff-301a36bbdb135d8cc270035d6ba0408e5f14291fd0a71e186f91bd0da01bf89bR11, Combobox doesn't seem to do so https://github.com/twilio-labs/paste/commit/2eb6dd74eb1cdc06079e64a711e71b93d4f9c823#diff-22b2d509777ec919245d83593056eb5d0c5ac23808f3b543b288425b985cb360R35. 

When I add a custom `element` prop to Combobox then the tests pass successfully. 

However please correct me if this is not the case here. Just wondering if this is something related to my project dependencies or a general issue.

**Reply by SiTaggart** [reply]
Clarifying question, is this Typescript throwing the error or PropTypes?

**Reply by mirjammalksoo** [reply]
This is PropTypes throwing the error when running unit tests. When compiling the project or viewing files, I don't get/see any errors.

**Reply by SiTaggart** [reply]
OK, yup. For what ever reason Typescript is slightly out of sync with PropTypes. I'll throw a fix out this morning for ya.

Sorry for the inconvenience.

**Reply by mirjammalksoo** [reply]
No problem, glad to get some clarification about the issue. 

Thank you!

**Comment by SiTaggart** [comment]
Identified the fix and a PR is up. https://github.com/twilio-labs/paste/pull/1824

We'll have this out as a patch by the end of the day.

**Comment by SiTaggart** [comment, marked as answer]
Paste Core 10.10.1 just went out. Let us know if that fixes your issues

**Comment by mirjammalksoo** [comment]
Yes this upgrade fixed all the previous PropTypes errors! Thank you again!
